src/core/scene/a-scene.js

```diff
@@ -294,9 +297,16 @@ class AScene extends AEntity {
         var xrMode = useAR ? 'immersive-ar' : 'immersive-vr';
         xrInit = this.sceneEl.systems.webxr.sessionConfiguration;
         return new Promise(function (resolve, reject) {
-          navigator.xr.requestSession(xrMode, xrInit).then(
+          var requestSession = useOfferSession ? navigator.xr.offerSession.bind(navigator.xr) : navigator.xr.requestSession.bind(navigator.xr);
+          self.usedOfferSession |= useOfferSession;
+          requestSession(xrMode, xrInit).then(
             function requestSuccess (xrSession) {
               self.xrSession = xrSession;
+
+              if (useOfferSession) {
+                self.usedOfferSession = false;
```

Comment:
Probably can always set this to false here?

Reply:
without checking useOfferSession

Reply:
I believe the regular requestSession API will then also set this to false. At that point we'll be in a bad state because the offersession promise is still pending.